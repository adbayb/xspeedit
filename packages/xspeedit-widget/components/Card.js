import React from "react";
import PropTypes from "prop-types";

import { BREAKPOINT } from "../constants/index";

const Card = ({ children, Menu }) => {
	return (
		<article>
			{children && <div>{children}</div>}
			{Menu && <footer>{Menu}</footer>}
			<style jsx>{`
				article {
					display: flex;
					flex: 1;
					flex-direction: column;
					background-color: #2d2e31;
					box-shadow: 3px -1px 6px 0px rgba(0, 0, 0, 0.3);
					overflow: hidden;
				}
				div,
				footer {
					display: flex;
					position: relative;
				}
				footer {
					flex: 2;
				}
				div {
					flex: 12;
				}

				@media (max-width: ${BREAKPOINT}px) {
					article {
						align-self: stretch;
					}
				}
				@media (min-width: ${BREAKPOINT}px) {
					article {
						max-width: 400px;
						height: 600px;
					}
				}
			`}</style>
		</article>
	);
};

Card.propTypes = {
	children: PropTypes.node.isRequired,
	Menu: PropTypes.node.isRequired
};

export default Card;
