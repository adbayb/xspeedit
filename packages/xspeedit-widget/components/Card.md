React component example:

```js
<Card Menu={<div>TITITI</div>}>TOTO</Card>;
```
