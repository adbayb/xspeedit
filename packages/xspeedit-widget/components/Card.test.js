import React from "react";
import Card from "./Card.js";

describe("Card", () => {
	it("should render correctly", () => {
		// GIVEN: notre composant:
		const tree = shallow(<Card Menu={<div>Menu</div>}> Content </Card>);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});
});
