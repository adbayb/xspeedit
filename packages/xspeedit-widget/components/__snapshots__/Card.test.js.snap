// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<article
  className="jsx-2521222757"
>
  <div
    className="jsx-2521222757"
  >
     Content 
  </div>
  <footer
    className="jsx-2521222757"
  >
    <div>
      Menu
    </div>
  </footer>
  <JSXStyle
    css="article.jsx-2521222757{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;background-color:#2d2e31;box-shadow:3px -1px 6px 0px rgba(0,0,0,0.3);overflow:hidden;}div.jsx-2521222757,footer.jsx-2521222757{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;position:relative;}footer.jsx-2521222757{-webkit-flex:2;-ms-flex:2;flex:2;}div.jsx-2521222757{-webkit-flex:12;-ms-flex:12;flex:12;}@media (max-width:768px){article.jsx-2521222757{-webkit-align-self:stretch;-ms-flex-item-align:stretch;align-self:stretch;}}@media (min-width:768px){article.jsx-2521222757{max-width:400px;height:600px;}}"
    styleId="2521222757"
  />
</article>
`;
